smart_open/s3.py

```diff
@@ -281,7 +281,8 @@ def read(self, size=-1):
         # Fill our buffer to the required size.
         #
         # logger.debug('filling %r byte-long buffer up to %r bytes', len(self._buffer), size)
-        self._fill_buffer(size)
+        fill_size = size if size > DEFAULT_BUFFER_SIZE else DEFAULT_BUFFER_SIZE
```

Comment:
```suggestion
        fill_size = max(size, DEFAULT_BUFFER_SIZE)
```

Reply:
Could you please apply this change?

I get "You do not have permission to push to this repository."